R/assertions.R

```diff
@@ -783,46 +784,66 @@ assert_s3_class <- function(arg, cls,
 #'   assert_list_of(list, "numeric", named = TRUE)
 #' }
 #' try(example_fun2(list(1, 2, 3, d = 4)))
-assert_list_of <- function(arg, class, named = FALSE, optional = TRUE) {
-  assert_character_scalar(class)
+assert_list_of <- function(arg, cls,
+                           named = FALSE,
+                           optional = TRUE,
+                           arg_name = rlang::caller_arg(arg),
```

Comment:
Just a minor comment, should we import the `caller_arg()` so we can just use it without `rlang::`, it looks like we're using that function in a few other admiraldev programs and have already imported quite a lot of the rlang functions
![image](https://github.com/pharmaverse/admiraldev/assets/112789549/42dbe808-efdf-4827-af28-a09d2967915d)

<img width="296" alt="image" src="https://github.com/pharmaverse/admiraldev/assets/112789549/38f0a856-6159-41a2-9bce-9224b97fb9df">

Reply:
That's an interesting point! Because these are default values for a function (which means they are user-facing), we would need to decide whether we want to leave the `rlang::` prefix or re-export the function. As we don't re-export other rlang functions, I think we can just leave the namespacing prefix in this case.

Reply:
I'm fine with leaving this as `rlang::` for the function arguments.